sogs/model.py

```diff
@@ -473,3 +511,35 @@ def get_message_deprecated(room_id, since, limit=256):
                 }
             )
     return msgs
+
+
+def ban_user(mod_user, room, ban_user):
+    """as a moderator mod_user ban a user from a room
+    return True if the user was banned, otherwise return False
+    """
```

Comment:
This would be better as a method on `User`, so that you do:

`user.ban(room, bad_user)`

Reply:
we need to be able to ban users from the bot api when we do not have a corresponding user in the database

Reply:
It might also be better to actual ban query into `Room` so that you can do:

`room.ban(bad_user)`

and

`user.ban(room, bad_user)`

would do the checks here then invoke `room.ban(bad_user)` if things are allowed.  (That, in particular, makes it easy to add bans from the CLI).

Reply:
i feel like inverting the user.ban calling would be better. 

```python
user.ban(mod=mod_user) # ban user from the entire server as mod
user.ban(room=room, mod=mod_user) # ban user from room as mod
user.ban(room=room) # ban user from room unconditonally 
user.ban() # ban user from entire server unconditionally
```

Reply:
another thought, we could do this: 

```python
with user.mod() as mod_user:
    mod_user.ban(user=other_guy, room=room)
```